src/main/java/uk/gov/hmcts/reform/unspec/config/ExternalTaskListenerConfiguration.java

```diff
@@ -17,7 +21,16 @@ public ExternalTaskListenerConfiguration(@Value("${feign.client.config.remoteRun
     @Bean
     public ExternalTaskClient client() {
         return ExternalTaskClient.create()
+            .addInterceptor(new ServiceAuthProvider())
             .baseUrl(baseUrl)
             .build();
     }
+
+    public class ServiceAuthProvider implements ClientRequestInterceptor {
+
+        @Override
+        public void intercept(ClientRequestContext requestContext) {
+            requestContext.addHeader(ServiceAuthFilter.AUTHORISATION, generate());
```

Comment:
We already have `ServiceTokenGeneratorConfiguration` which exposes `AuthTokenGenerator` bean. we can use that to generate token and pass it here. We don't need to implement `AuthTokenGenerator`